README.md

```diff
@@ -1 +1,56 @@
 # cartesi-paio
+
+
+## User transactions
+
+Users build an [EIP-712](https://eips.ethereum.org/EIPS/eip-712) signed transaction. A `SignedTransaction` consists of the pair `SigningMessage` and a `Signature`. They then submit it to a sequencer frontend.
+
+Note that this signed transaction contains the address of the destination dapp. It also contains a nonce (which is per application), and the maximum gas price this user is willing to pay _for DA_.
+
+
+## Sequencer frontend
+
+The frontend is the component in the sequencer that receives user transactions.
+Think that, in some frequency, this component yields a bunch of unordered `SignedTransaction`.
+
+There are many kinds of sequencer front end:
+* In a centralized sequencer, users send transactions to the sequencer through a “submit endpoint”.
+* In a based sequencer, users send transactions to a mempool/p2p-network, and the set of Ethereum block builders running some version of chorizo will pick these transactions up.
+* In Espresso, users send transactions to the Espresso network, which is then picked up by the sequencers, and finally the elected builder has the rights to sequence these transactions.
+
+
+## Sequencer batcher
+
+The batcher is the component in the sequencer that takes an unordered set of `SignedTransaction`s, and builds a `Batch`, consisting of an ordered set of `WireTransaction`s.
```

Comment:
Calling the transactions "unordered" is perhaps too much. It is just that other people may disagree on their exact order. But the sequencer received one after another.